### PR TITLE
map ip to country

### PR DESCRIPTION
**Pull request by claudiunicolaa**

### CODE REVIEW

**Comment by claudiunicolaa** [comment]
resolves https://github.com/curveball/a12n-server/issues/40

File: mysql-schema/034-userlog4.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+SET NAMES utf8mb4;
+START TRANSACTION;
+
+INSERT INTO changelog VALUES (34, UNIX_TIMESTAMP());
+
+ALTER TABLE user_log ADD
+  country VARCHAR(50) NULL;
```
Comment (evert) [inline mysql-schema/034-userlog4.sql]: Could we store the country as a 2-byte character string instead?

File: src/log/service.ts [hunk 2]
```diff
@@ -67,3 +71,11 @@ function isContext(ctx: any): ctx is Context {
   return ((<Context> ctx).ip !== undefined);
 
 }
+
+function getCountryByIp(ip: string)
```
Comment (evert) [inline src/log/service.ts]: I had a look in this library, and I think it's worth making a small adjustment.

The library will return a country, _or_ the string `?`. I don't think it's good to store `?` anywhere, so instead I think this function should have a condition for this and return `null` if a country could not be determined.

**Comment by evert** [comment]
Also, thank you for the contribution. It's definitely appreciated!

**Comment by evert** [comment]
This _might_ be a better alternative, as it packs in a database and is using a license that _is_ compatible with ours.

File: src/log/service.ts [hunk 3]
```diff
@@ -2,6 +2,8 @@ import { Context } from '@curveball/core';
 import db from '../database';
 import { User } from '../user/types';
 import { EventType, LogEntry } from './types';
+// @ts-ignore: No types for this yet
```
Comment (evert) [inline src/log/service.ts]: It looks like there is a `@types/geoip-lite` package, so this should not be needed when this is installed!

**Comment by evert** [comment]
#209 replaces this PR

**Pull request closed without merging**